Should be able to copy editable strong password on iOS

erynofwales opened the pull request:
#### f07a52b9573be1d5b638c719cbf84b5eaed88c5f
<pre>
Should be able to copy editable strong password on iOS

<a href="https://bugs.webkit.org/show_bug.cgi?id=242544">https://bugs.webkit.org/show_bug.cgi?id=242544</a>

rdar://96259620

Reviewed by Tim Horton and Wenson Hsieh.

Add a flag to EditorState to track when selection is in an autofilled and
viewable field. Consult this flag in WKContentView when we&apos;re deciding what
editing actions to allow.

Test: WebKit.CopyInAutoFilledAndViewablePasswordField

* Source/WebKit/Shared/EditorState.cpp:
(WebKit::EditorState::encode const):
(WebKit::EditorState::decode):
    Implement transmitting the new flag over the IPC interface

* Source/WebKit/Shared/EditorState.h:
    Add selectionIsInAutoFilledAndViewableField to EditorState
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView canPerformActionForWebView:withSender:]):
    Check the new flag when deciding whether to add the Copy item to the list of
    available actions

* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::editorState const):
    Set the new flag based on whether the selection is in an AutoFilled and
    Viewable field

* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKContentViewEditingActions.mm:
    Add a new API test to exercise this fix.

* metadata/contributors.json:
    Add myself to the contributors file

Canonical link: <a href="https://commits.webkit.org/252351@main">https://commits.webkit.org/252351@main</a>
</pre>

whsieh commented on Source/WebKit/WebProcess/WebPage/WebPage.cpp:
```diff
@@ -1251,6 +1251,7 @@ EditorState WebPage::editorState(ShouldPerformLayout shouldPerformLayout) const
     if (result.selectionIsRange) {
         auto selectionRange = selection.range();
         result.selectionIsRangeInsideImageOverlay = selectionRange && ImageOverlay::isInsideOverlay(*selectionRange);
+        result.selectionIsInAutoFilledAndViewableField = selection.isInAutoFilledAndViewableField();
```
(On second glance, I would probably call this `selectionIsRangeInAutoFilledAndViewableField`, to be more exact)

erynofwales commented on metadata/contributors.json:
```diff
@@ -2421,6 +2421,15 @@
+      "nicks" : [
+         "erynofwales"
+      ]
+   },
```
Added myself to the contributors file in this commit. 🙂

erynofwales committed:
Should be able to copy editable strong password on iOS

https://bugs.webkit.org/show_bug.cgi?id=242544

rdar://96259620

Reviewed by Tim Horton and Wenson Hsieh.

Add a flag to EditorState to track when selection is in an autofilled and
viewable field. Consult this flag in WKContentView when we're deciding what
editing actions to allow.

Test: WebKit.CopyInAutoFilledAndViewablePasswordField

* Source/WebKit/Shared/EditorState.cpp:
(WebKit::EditorState::encode const):
(WebKit::EditorState::decode):
    Implement transmitting the new flag over the IPC interface

* Source/WebKit/Shared/EditorState.h:
    Add selectionIsInAutoFilledAndViewableField to EditorState
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView canPerformActionForWebView:withSender:]):
    Check the new flag when deciding whether to add the Copy item to the list of
    available actions

* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::editorState const):
    Set the new flag based on whether the selection is in an AutoFilled and
    Viewable field

* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKContentViewEditingActions.mm:
    Add a new API test to exercise this fix.

* metadata/contributors.json:
    Add myself to the contributors file

Canonical link: https://commits.webkit.org/252351@main
